src/chunker.c

```diff
@@ -8,13 +8,70 @@
  * optimization. See RFC1950 for details. */
 #define BYTES_PUSH_MAX 5552U
 
+/* Unless otherwise specified, take this as the default average chunk size */
+#define DEFAULT_CHUNK_SIZE_AVG (16 * 1024)
+
+static bool number_is_prime(unsigned int n) {
+        unsigned int i;
+
+        for (i = 2; i < n; i++)
+                if (n % i == 0)
+                        return false;
+
+        return true;
+}
+
+int ca_chunker_set_avg_size(CaChunker *c, size_t avg) {
+        size_t delta, closest;
+
+        assert(c);
+
+        /* Find an average block size that is close to the requested number but is prime,
+         * because this number is used in the modulo operation that determines the chunk
+         * break points and we don't want to waste any bits there. */
+
+        if (c->a > 1)
```

Comment:
is this test really suitable? a might become 0 eventually again, no?

better test would be to check window_size != 0...